src/Raven.Server/Documents/Handlers/Processors/Documents/AbstractDocumentHandlerProcessorForGet.cs

```diff
@@ -323,119 +329,106 @@ await using (AsyncBlittableJsonTextWriter.Create(context, RequestHandler.Respons
 
     protected abstract ValueTask<DocumentsResult> GetDocumentsImplAsync(TOperationContext context, long? etag, StartsWithParams startsWith, string changeVector);
 
-    private async Task<StringValues> GetIdsFromRequestBody(TOperationContext context)
-    {
-        var docs = await context.ReadForMemoryAsync(RequestHandler.RequestBodyStream(), "docs");
-        if (docs.TryGet("Ids", out BlittableJsonReaderArray array) == false)
-            Web.RequestHandler.ThrowRequiredPropertyNameInRequest("Ids");
-
-        var idsAsStrings = new string[array.Length];
-
-        for (int i = 0; i < array.Length; i++)
-        {
-            idsAsStrings[i] = array.GetStringByIndex(i);
-        }
-
-        return new StringValues(idsAsStrings);
-    }
-
-    private RevisionIncludeField GetRevisionsToInclude()
+    private static RevisionIncludeField GetRevisionsToInclude(QueryStringParameters parameters)
     {
-        var revisionsByChangeVectors = RequestHandler.GetStringValuesQueryString("revisions", required: false);
-        var revisionByDateTimeBefore = RequestHandler.GetStringValuesQueryString("revisionsBefore", required: false);
-
-        if (revisionsByChangeVectors.Count == 0 && revisionByDateTimeBefore.Count == 0)
+        if (parameters.Revisions == null && parameters.RevisionsBefore == null)
             return null;
 
         var rif = new RevisionIncludeField();
 
-        if (DateTime.TryParseExact(revisionByDateTimeBefore.ToString(), DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal, out var dateTime))
+        if (parameters.RevisionsBefore.HasValue && DateTime.TryParseExact(parameters.RevisionsBefore.Value.Span, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal, out var dateTime))
             rif.RevisionsBeforeDateTime = dateTime.ToUniversalTime();
 
-        foreach (var changeVector in revisionsByChangeVectors)
-            rif.RevisionsChangeVectorsPaths.Add(changeVector);
+        if (parameters.Revisions != null)
+        {
+            foreach (var changeVector in parameters.Revisions)
+                rif.RevisionsChangeVectorsPaths.Add(changeVector.ToString());
```

Comment:
Can `changeVector` be null? Previously we didn't call `.ToString()` on it.

Reply:
It's ReadOnlyMemory<char> now, didn't want to go too deep for all the paths, so previously it was StringValues and there was implicit conversion to string